core/trino-main/src/main/java/io/trino/operator/join/PartitionedConsumption.java

```diff
@@ -78,7 +85,7 @@ private PartitionedConsumption(
         for (Integer partitionNumber : partitionNumbers) {
             Partition<T> partition = new Partition<>(consumersCount, partitionNumber, loader, partitionActivator, disposer);
             partitions.add(partition);
-            partitionActivator = partition.released;
```

Comment:
I think `partition.released` can now be removed and `disposer` called directly in `PartitionedConsumption.Partition#release`